lib/ansible/modules/system/aix_lvg.py

```diff
@@ -0,0 +1,438 @@
+        if rc != 0:
+            module.fail_json(msg="Failing to execute %s command." % lsvg_cmd)
+
+        pvs_to_remove = []
+        for line in current_pvs.splitlines()[2:]:
+            pvs_to_remove.append(line.split()[0])
+
+        reduce_msg = "Volume group %s removed." % vg
+    else:
+        pvs_to_remove = pvs
+        reduce_msg = (
+            "Physical volume(s) %s removed from Volume group %s." % (
+                " ".join(pvs_to_remove), vg))
+
+    # Reduce volume group.
+    if len(pvs_to_remove) > 0:
+        changed = True
+        msg = ''
+
+        if not module.check_mode:
+            reducevg_cmd = module.get_bin_path("reducevg", True)
+            rc, _, err = module.run_command(
+                "%s -df %s %s" % (reducevg_cmd, vg, " ".join(pvs_to_remove)))
+
+            if rc == 0:
+
+                return changed, reduce_msg
+
+            else:
+                module.fail_json(
+                    msg="Unable to remove %s." % vg, rc=rc, err=err)
+
+        return changed, msg
+
+    else:
+        changed = False
+        msg = "No physical volumes to remove."
+
+        return changed, msg
+
+
+def state_vg(module, vg, state, vg_validation):
+    vg_state, msg = vg_validation
+
+    if vg_state is None:
+        module.fail_json(msg=msg)
+
+    if state == 'varyon':
+        if vg_state is False:
+
+            changed = True
+            msg = ''
+            if not module.check_mode:
+                varyonvg_cmd = module.get_bin_path("varyonvg", True)
+                rc, varyonvg_out, err = module.run_command(
+                    "%s %s" % (varyonvg_cmd, vg))
+                if rc != 0:
+                    module.fail_json(msg="Varyonvg failed.", rc=rc, err=err)
+                else:
+                    msg = "Varyon volume group %s completed." % vg
+
+            return changed, msg
+
+        else:
+            changed = False
+
+            return changed, msg
+
+    elif state == 'varyoff':
+        if vg_state is True:
+
+            changed = True
+            msg = ''
+            if not module.check_mode:
+                varyonvg_cmd = module.get_bin_path("varyoffvg", True)
+                rc, varyonvg_out, err = module.run_command(
+                    "%s %s" % (varyonvg_cmd, vg))
+                if rc != 0:
+                    module.fail_json(msg="Varyoffvg failed.", rc=rc, err=err)
+                else:
+                    msg = "Varyoff volume group %s completed." % vg
+
+                    return changed, msg
+
+            return changed, msg
+
+        else:
+            changed = False
+
+            return changed, msg
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            force=dict(type="bool", default=False),
+            pp_size=dict(type="int", default=None),
+            pvs=dict(type="list", default=None),
+            state=dict(choices=["absent", "present", "varyon", "varyoff"],
```

Comment:
This should be:

```python
            state=dict(
                choices=["absent", "present", "varyon", "varyoff"],
                default="present"),
```

The same for `vg_type` bellow.

Reply:
Regarding double/single quotes. First, is hard to change some habits. :)

I suppose we are talking regarding ```msg = ''```, right? I was in doubt about it and I checked module mount.py and I got more confused now.